software/tinyclr/tutorials/networking/spwf04sx-wifi.md

```diff
@@ -0,0 +1,446 @@
+
+### SPWF04SxInterface()
+
+`SPWF04SxInterface(SpiDevice spi, GpioPin irq, GpioPin reset)`. Used to create a new instance of the Wi-Fi driver. Takes the SPI device and the IRQ and RESET GPIO pins as arguments. Has no return value.
+
+### TurnOff()
+
+`TurnOff()`. Turns off the Wi-Fi module. Takes no arguments and has no return value.
+
+### TurnOn()
+
+`TurnOn()`. Turns on the Wi-Fi module. Takes no arguments and has no return value.
+
+### WriteSocket()
+
+`WriteSocket(int socket, byte[] data)`. Writes data to a socket client. Arguments are an integer ID number for the socket and a byte array of data. The socket id is returned by the `OpenSocket()` method when the socket is created.
+
+`WriteSocket(int socket, byte[] data, int offset, int count)`. Same as above but also takes integer arguments for the offset into the data array and a count for the number bytes to send.
+
+## SPWF04Sx Events
+
+### IndicationReceived
+
+Fires an event when a Wi-Fi indication (WIND) message is received.
+
+### ErrorReceived
+
+Fires an event when a Wi-Fi error message is received.
+
+## SPWF04Sx Properties
+
+### ForceSocketsTls
+
+Boolean property that can be used to ensure a socket is opened as a secure TLS socket. For example, executing `wifi.ForceSocketsTls = true;` before `OpenSocket()` will cause the socket to be opened as a TLS Socket.
+
+### ForceSocketsTlsCommonName
+
+String property used to specify the common name as found in the websites certificate. The common name is listed in the `Subject Alternative Name` field of the certificate.
+
+### State
+
+Read only property of type `SPWF04SxWiFiState` that reflects the current state of the Wi-Fi module. For example:
+
+`if (wifi.State == SPWF04SxWiFiState.HardwareFailure) Debug.WriteLine("Hardware Failure");`
+
+The possible states are: `HardwarePowerUp`, `HardwareFailure`, `RadioTerminatedByUser`, `RadioIdle`, `ScanInProgress`, `ScanComplete`, `JoinInProgress`, `Joined`, `AccessPointStarted`, `HandshakeComplete`, and `ReadyToTransmit`.
+
+## Sample Code
+
+The following sample code will work on the FEZ and the UC5550 if they have Wi-Fi. Just comment out the code for the board you are not using (works as is on the FEZ).
+
+```csharp
+using GHIElectronics.TinyCLR.Devices.Gpio;
+using GHIElectronics.TinyCLR.Devices.Spi;
+using GHIElectronics.TinyCLR.Drivers.STMicroelectronics.SPWF04Sx;
+using GHIElectronics.TinyCLR.Pins;
+using System;
+using System.Diagnostics;
+using System.Net;
+using System.Net.NetworkInterface;
+using System.Net.Sockets;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+
+namespace WiFi {
+    public static class Program {
+        private static GpioPin led1;
+        private static GpioPin btn1;
+        private static SPWF04SxInterface wifi;
+
+        public static void Main() {
+            var cont = GpioController.GetDefault();
+
+            //FEZ
```

Comment:
Get rid of separate FEZ and UC5550, just open the pins and leave a spot for the user to fill in their pins